### PR TITLE
fix: avoid hardcoded versions for Thanos components

### LINKED ISSUE #250: Avoid hardcoded versions for Thanos components
The observability operator hard-codes the versions used for the Thanos sidecar and querier. The current value is v0.24.0 which is rather old.

For the Thanos sidecar, we could leave the fieds unset and the Prometheus operator would use whatever is the latest Thanos version when the operator was released. For the querier, we could rely on [pkg/operator/defaults.go](https://github.com/prometheus-operator/prometheus-operator/blob/e4e27052f57040f073c6c1e4aedaecaaec77d170/pkg/operator/defaults.go#L28) from `github.com/rhobs/obo-prometheus-operator`.

https://github.com/rhobs/observability-operator/blob/c2ecb858856e68d1cf2f376b0b875410ef74d8ed/pkg/controllers/monitoring/monitoring-stack/components.go#L179-L182

https://github.com/rhobs/observability-operator/blob/c2ecb858856e68d1cf2f376b0b875410ef74d8ed/pkg/controllers/monitoring/thanos-querier/components.go#L71

### PR DESCRIPTION
**Pull request by slashpai**
Fixes #250

### CODE REVIEW

File: pkg/controllers/monitoring/monitoring-stack/components.go [hunk 1]
```diff
@@ -181,8 +182,8 @@ func newPrometheus(
 			RuleSelector:          prometheusSelector,
 			RuleNamespaceSelector: ms.Spec.NamespaceSelector,
 			Thanos: &monv1.ThanosSpec{
-				BaseImage: stringPtr("quay.io/thanos/thanos"),
-				Version:   stringPtr("v0.24.0"),
+				BaseImage: stringPtr(po.DefaultThanosBaseImage),
```
Comment (simonpasquier) [inline pkg/controllers/monitoring/monitoring-stack/components.go:185]: we shouldn't use baseImage which is deprecated but `image` instead.

**Comment by slashpai** [comment]
Facing issue with `go mod tidy`

```
░▒▓    ~/gi/s/observability-operator  on   thanos_version *2  go mod tidy
go: finding module for package github.com/prometheus/prometheus/model/labels
go: finding module for package github.com/prometheus/prometheus/model/rulefmt
github.com/rhobs/observability-operator/pkg/controllers/monitoring/monitoring-stack imports
        github.com/rhobs/obo-prometheus-operator/pkg/operator imports
        github.com/prometheus/prometheus/model/rulefmt: package github.com/prometheus/prometheus/model/rulefmt provided by github.com/prometheus/prometheus at latest version v0.46.0 but not at required version v1.8.2-0.20211105201321-411021ada9ab
github.com/rhobs/observability-operator/pkg/controllers/monitoring/monitoring-stack imports
        github.com/rhobs/obo-prometheus-operator/pkg/operator imports
        github.com/rhobs/obo-prometheus-operator/pkg/namespacelabeler imports
        github.com/prometheus/prometheus/model/labels: package github.com/prometheus/prometheus/model/labels provided by github.com/prometheus/prometheus at latest version v0.46.0 but not at required version v1.8.2-0.20211105201321-411021ada9ab
```

Upon troubleshooting it appears that unable to bump prometheus module to latest since https://github.com/kubernetes-sigs/instrumentation-tools/blob/master/go.mod#L15 depends on older version of module

**Comment by slashpai** [comment]
https://github.com/kubernetes-sigs/instrumentation-tools/pull/8 created for bumping go version and dependencies

**Comment by jan--f** [comment]
/close
#250 was fixed elsewhere.

**Pull request closed without merging**